### PR TITLE
[refine](Column)Disallow implicit conversion of ColumnPtr to IColumn*

### PR DESCRIPTION
**Pull request by Mryange**
### What problem does this PR solve?

Previously, we allowed ColumnPtr to be directly converted to Column*:  
```C++
    ColumnPtr column;  
    const IColumn* ptr = column;  
```
This can easily cause confusion.  
For example, in the following code: 
```C++
    ColumnPtr column;  
    const auto& const_column = check_and_get_column<ColumnConst>(column);  
```
The matched function is:  
```C++
template <>  
const doris::vectorized::ColumnConst* check_and_get_column<doris::vectorized::ColumnConst>(  
        const IColumn* column)  
```
However, the actual type of const_column is:  
```C++
const doris::vectorized::ColumnConst* const&
```

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [x] No need to test or manual test. Explain why:
        - [x] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

File: be/src/vec/common/cow.h [hunk 1]
```diff
@@ -203,7 +203,12 @@ class COW {
 
         operator bool() const { return t != nullptr; }
 
-        operator T*() const { return t; }
+#ifndef BE_TEST
+        [[deprecated("Implicit conversion should not be used. Please use get() explicitly.")]]
+#endif
+        operator T*() const {
```
Comment (yiguolei) [inline be/src/vec/common/cow.h:209]: delete this code， no need to give warnings

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

**Review by zhangstar333 — APPROVED** [review]

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**